autoload/silicon.vim

```diff
@@ -88,12 +88,31 @@ let s:default = {
 
 call s:configure('g:silicon', s:default)
 
+if !exists('g:silicon_default_file_pattern')
+  " Examples:
+  "   ~/images/silicon-{time:%Y-%m-%d-%H%M%S}.png
+  "   ~/images/silicon-{time:%s}.png
+  "
+  let g:silicon_default_file_pattern = ''
+endif
+
 " Silicon bindings
 fun! s:cmd(argc, argv)
   let cmd = ['silicon']
   " Output method
   if a:argc == 0
-    if empty(executable('xclip'))
+    if len(g:silicon_default_file_pattern) > 0
+      let path = g:silicon_default_file_pattern
+      let path = substitute(g:silicon_default_file_pattern,
+            \ '{time:\(.\{-}\)}', {match -> strftime(match[1])}, 'g')
```

Comment:
This could be a named `s:RenderFilePattern` function or something, which not only allows `{time:}`, but maybe also `{filename:}`, `{line:}`, `{col:}` placeholders. Though I'm not sure about the details on the filename -- escaped full path, basename, or maybe full path, so that `/foo/bar/code.rb` turns into `/foo/bar/code.png`? 

A timestamp seems like the simplest way to go, which is I why I stuck to that.